cow_amm/cow_amm_lp_and_tvl_4047078

```diff
@@ -0,0 +1,48 @@
+with 
+cow_amm_pool as (
+select blockchain, address, token_1_address, token_2_address
+from query_3959044 q
+where address={{cow_amm_pool}}
+)
+, get_lp_balance as (
+    select date(evt_block_time) as day, 
+    sum(case when "from" = 0x0000000000000000000000000000000000000000 THEN (value/1e18) ELSE -(value/1e18) END) as lp_supply
+    from erc20_ethereum.evt_transfer
+    where contract_address = {{cow_amm_pool}}
+    and ("from" = 0x0000000000000000000000000000000000000000 or "to" = 0x0000000000000000000000000000000000000000)
+    group by 1
+)
+
+, total_lp as (
+select day, lp_supply, sum(lp_supply) over (order by day) as total_lp
+from get_lp_balance
+)
+
+, get_tvl as (
+select 
+  date(time) as day, 
+  time, 
+  blockchain, 
+  cow_amm_address, 
+  token_1_balance_usd+token_2_balance_usd as tvl,
+  RANK() OVER (PARTITION BY date(time) ORDER BY time DESC) AS tvl_rank
+from  dune.cowprotocol.result_balancer_cow_amm_base_query_v_2
```

Comment:
Are we sure we want to rely on this materialized view for TVL? Is it updating reliably now (last time I checked it was stuck many days in the past)? 

Have you tried using something like
```sql
select day, balance
from tokens_ethereum.balances_daily
where address = {{ cow_amm }}
and token_address = {{ token }}
and blockchain = 'ethereum'
```

to get the TVL?

Reply:
Yes, I did check that table. It is way heavier and takes longer to compute the values (takes ~4mins to get those values), hence, I was trying to avoid it, but if there is more confidence I can update it. Ultimately it gives the same values. 

In the past few days, the table (dune.cowprotocol.result_balancer_cow_amm_base_query_v_2) didn't have any issues, but I did not find a way to check the historical view of the refreshes.

Reply:
Makes sense to not use `balances_daily` if it is too slow. I wonder if we can use a similar approach as for the LP tokens (summing up transfers of token1 and token2 in and out of this pool and multiply it with the price at the end of each day) to get the final balance efficiently without relying on the materialized view.